antarest/study/storage/output_service.py

```diff
@@ -0,0 +1,531 @@
+from antarest.core.requests import RequestParameters
+from antarest.core.serde.json import to_json
+from antarest.core.tasks.model import TaskListFilter, TaskResult, TaskStatus, TaskType
+from antarest.core.tasks.service import ITaskNotifier
+from antarest.core.utils.archives import ArchiveFormat
+from antarest.core.utils.utils import StopWatch
+from antarest.study.business.aggregator_management import (
+    AggregatorManager,
+    MCAllAreasQueryFile,
+    MCAllLinksQueryFile,
+    MCIndAreasQueryFile,
+    MCIndLinksQueryFile,
+)
+from antarest.study.model import ExportFormat, Study, StudyDownloadDTO, StudySimResultDTO
+from antarest.study.service import StudyService
+from antarest.study.storage.rawstudy.model.filesystem.matrix.matrix import MatrixFrequency
+from antarest.study.storage.rawstudy.model.filesystem.root.output.simulation.mode.mcall.digest import (
+    DigestSynthesis,
+    DigestUI,
+)
+from antarest.study.storage.study_download_utils import StudyDownloader, get_output_variables_information
+from antarest.study.storage.utils import assert_permission, is_managed, is_output_archived, remove_from_cache
+from antarest.worker.archive_worker import ArchiveTaskArgs
+
+logger = logging.getLogger(__name__)
+
+
+class OutputService:
+    def __init__(self, study_service: StudyService):
+        self.study_service = study_service
+
+    def get_digest_file(self, study_id: str, output_id: str, params: RequestParameters) -> DigestUI:
+        study = self.study_service.get_study(study_id)
+        assert_permission(params.user, study, StudyPermissionType.READ)
+        file_study = self.study_service.storage_service.get_storage(study).get_raw(study)
+        digest_node = file_study.tree.get_node(url=["output", output_id, "economy", "mc-all", "grid", "digest"])
+        assert isinstance(digest_node, DigestSynthesis)
+        return digest_node.get_ui()
+
+    @staticmethod
+    def _get_output_archive_task_names(study: Study, output_id: str) -> tuple[str, str]:
+        return (
+            f"Archive output {study.id}/{output_id}",
+            f"Unarchive output {study.name}/{output_id} ({study.id})",
+        )
+
+    def unarchive_output(
+        self,
+        study_id: str,
+        output_id: str,
+        keep_src_zip: bool,
+        params: RequestParameters,
+    ) -> Optional[str]:
+        study = self.study_service.get_study(study_id)
+        assert_permission(params.user, study, StudyPermissionType.READ)
+        self.study_service.assert_study_unarchived(study)
+
+        output_path = Path(study.path) / "output" / output_id
+        if not is_output_archived(output_path):
+            if not output_path.exists():
+                raise OutputNotFound(output_id)
+            raise OutputAlreadyUnarchived(output_id)
+
+        archive_task_names = OutputService._get_output_archive_task_names(study, output_id)
+        task_name = archive_task_names[1]
+
+        study_tasks = self.study_service.task_service.list_tasks(
```

Comment:
the task service should be an explicit dependency of this one, if it uses it directly, it should not be used from the study service, it's an implementation detail of that one